pilot/pkg/xds/filters/filters.go

```diff
@@ -360,6 +361,63 @@ var (
 			}),
 		},
 	}
+	// BuildConnectOriginalDstFallbackFilter builds a filter that sets the original destination
+	// filter state. It uses the pod IP and extracts the port from the :authority header.
+	// This ensures the filter state is populated with a valid IP:port for routing.
+	BuildConnectOriginalDstFallbackFilter = func(podIP string) *hcm.HttpFilter {
+		// Template-level concatenation approach - avoid using CEL + operator
+		// Logic: If filter state key exists, use its value as-is
+		// If filter state key doesn't exist, use fallback IP with extracted port
+		// Note: Use 2 backslashes in Go raw string literal - protobuf marshaling will double them to 4 in JSON
+		// The JSON will show \\\\1 (4 backslashes), and CEL will interpret as \\1 for the regex backreference
```

Comment:
Can you use Go backticks "``" for string literals? Less escaping

Reply:
will check